### PR TITLE
Refactor test sdk props and target files

### LINKED ISSUE #1827: Microsoft.Net.Test.Sdk specific tfm targets should not contain TargetFrameworkIdentifier checks
Right now the `Microsoft.Net.Test.Sdk.targets` file contains conditions that check on the `TargetFrameworkIdentifier`. That makes sense for the file that is placed under `buildMultiTargeting` but not for the files that are under specific tfm directories, i.e. `netcoreapp1.0`.

Removing these conditions and code paths that aren't related to the specific tfms would help us in corefx. That would help us with our current testing strategy running netstandard test assemblies against different tfms without cross-compiling.

A solution here would be to factor out common code into a file and import it by the different tfm targets files. If you are ok with my proposal I can quickly spin up PR which addresses this.

cc @ericstj

### PR DESCRIPTION
**Pull request by ViktorHofer**
Fixes https://github.com/Microsoft/vstest/issues/1827

1. Adding `MSBuildAllProjects` to all props and targets files to enable incremental updates when these files change.
2. Splitting up netfx and netcoreapp related code to get rid of the TargetFrameworkIdentifier checks. That supports our testing of netstandard assembly that we do in corefx.
3. Distribute entrypoint assets only for netcoreapp as netfx doesn't use that functionality.
4. Align copyright header which was different between the props and target file and fix few code styling issues.

cc @mayankbansal018

### CODE REVIEW

**Review by singhsarab — APPROVED** [review]
LGTM

File: src/package/nuspec/netcoreapp/Microsoft.NET.Test.Sdk.targets [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+           created a backup copy.  Incorrect changes to this file will make it
+           impossible to load or build your projects from the command-line or the IDE.
+
+ Copyright (c) .NET Foundation. All rights reserved. 
+ ***********************************************************************************************
+-->
+
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <PropertyGroup>
```
Comment (mayankbansal018) [inline src/package/nuspec/netcoreapp/Microsoft.NET.Test.Sdk.targets]: This is already defined in props file, is there a reason we need to redefine it here, if not, can you please remove it.

File: src/package/nuspec/netfx/Microsoft.NET.Test.Sdk.targets [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+           impossible to load or build your projects from the command-line or the IDE.
+
+ Copyright (c) .NET Foundation. All rights reserved. 
+ ***********************************************************************************************
+-->
+
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <PropertyGroup>
+    <MSBuildAllProjects>$(MSBuildAllProjects);$(MSBuildThisFileFullPath)</MSBuildAllProjects>
```
Comment (mayankbansal018) [inline src/package/nuspec/netfx/Microsoft.NET.Test.Sdk.targets]: Same as above.

File: src/package/nuspec/Microsoft.NET.Test.Sdk.props [hunk 3]
```diff
@@ -1,20 +1,27 @@
...
+           impossible to load or build your projects from the command-line or the IDE.
+ 
+ Copyright (c) .NET Foundation. All rights reserved. 
+ ***********************************************************************************************
 -->
 
 <Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <PropertyGroup>
+    <MSBuildAllProjects>$(MSBuildAllProjects);$(MSBuildThisFileFullPath)</MSBuildAllProjects>
```
Comment (mayankbansal018) [inline src/package/nuspec/Microsoft.NET.Test.Sdk.props]: This was earlier conditioned only for .netcore scenario , I'm not sure what it does, but do we need it for netfx?

**Review by mayankbansal018 — CHANGES_REQUESTED** [review]

**Comment by ViktorHofer** [comment]
MSBuildAllProjects adds the file as a msbuild input, means if you change that file it triggers a rebuild. You should have that in all your msbuild files, otherwise your changes won't reach customers until you do a clean build.

https://blogs.msdn.microsoft.com/msbuild/2005/09/26/how-to-ensure-changes-to-a-custom-target-file-prompt-a-rebuild/

**Comment by ViktorHofer** [comment]
When this is merged, can we ship an updated package of this either on nuget or a myget feed so that we can consume the changes in corefx?

**Review by mayankbansal018 — APPROVED** [review]

**Comment by ViktorHofer** [comment]
@mmitche PR is stuck

**Comment by mmitche** [comment]
@dotnet-bot test this please

**Comment by mmitche** [comment]
Can you close and reopen this PR?

**Comment by mayankbansal018** [comment]
@mmitche closed & reopened it.

**Comment by mmitche** [comment]
Can you confirm dotnet-bot still has write access to the repo?

**Comment by mayankbansal018** [comment]
@mmitche , I think we know what broke this. @cltshivash can you please write access to dotnet bot on vstest, & testfx repo.

**Comment by ViktorHofer** [comment]
Ok great, CI works again. The `Create-NugetPackages` script part needs some updates. I will do that later.

**Comment by ViktorHofer** [comment]
Ok fixed.

**Comment by ViktorHofer** [comment]
Thanks. Asking again:

> When this is merged, can we ship an updated package of this either on nuget or a myget feed so that we can consume the changes in corefx?

**Comment by singhsarab** [comment]
We are planning a release early next week, will keep you updated.
If you plan to do some testing, you can use the package number *16.0.0-preview-20181116-02* that is available on the myget feed.

**Comment by ViktorHofer** [comment]
Early next week works fine for me. Thanks!

**Pull request merged**